File: 1. 단계별로_풀어보기/20~29/23. 그리디 알고리즘 1/1. 11047.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

N, K = map(int, input().split())
coinArr = list(int(input()) for i in range(N))

coinArr.reverse()
result = 0

for coin in coinArr:
  div = K // coin
  result += div
  K -= div * coin

output(f"{result}")

File: 1. 단계별로_풀어보기/20~29/23. 그리디 알고리즘 1/1. 11047.js
// 입력
const fs = require('fs')
const inputArr = fs.readFileSync(0).toString().trim().split('\n')

// 입력 파싱
const [N, K] = inputArr[0].trim().split(' ').map(Number)
const coinArr = inputArr.slice(1).map(Number)

// 내림차순으로 변경
coinArr.reverse()

// 결과 카운팅할 변수
let remain = K
let result = 0
coinArr.forEach(coin => {
  const div = Math.floor(remain / coin)
  result += div
  remain -= div * coin
})

// 결과 출력
process.stdout.write(`${result}`)
